File: code/generate_strategies/Direct/java/function_catboost_params.java
package com.example;

import ai.catboost.CatBoostModel;
import ai.catboost.CatBoostPredictions;

public class CatBoostParamFetcher {
    public String get_model_params(int iterations, int depth, float learning_rate) {
        class CatBoostParamFetcher {
            public String getModelParams(int iterations, int depth, float learning_rate) {
                // 在Java中，我们使用CatBoost4j-Prediction库，它主要用于模型预测
                // 由于Java库的限制，我们直接返回传入的参数
                return String.format("Model Params - Iterations: %d, Depth: %d, Learning Rate: %.2f",
                        iterations, depth, learning_rate);
            }
        }

        CatBoostParamFetcher paramFetcher = new CatBoostParamFetcher();
        return paramFetcher.getModelParams(iterations, depth, learning_rate);
    }
}

File: manual/code/test_suites/cpp/FunctionBuildTest/src/function_catboost_params.cpp
#include <string>
#include <sstream>
#include <iomanip>

class CatBoostParamFetcher {
public:
    std::string getModelParams(int iterations, int depth, float learning_rate) {
        std::ostringstream oss;
        oss << "Model Params - Iterations: " << iterations << ", Depth: " << depth << ", Learning Rate: " << std::fixed << std::setprecision(2) << learning_rate;
        return oss.str();
    }
};